dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/S3Utils.java

```diff
@@ -175,21 +175,30 @@ public String getFileName(ResourceType resourceType, String tenantCode, String f
     }
 
     @Override
-    public void download(String tenantCode, String srcFilePath, String dstFile, boolean deleteSource, boolean overwrite) throws IOException {
+    public void download(String tenantCode, String srcFilePath, String dstFilePath, boolean deleteSource, boolean overwrite) throws IOException {
+        File dstFile = new File(dstFilePath);
+        if (dstFile.exists()) {
+            if (dstFile.isDirectory()) {
+                logger.error("destination file must be a file, but {} is a folder", dstFilePath);
+                throw new IOException("destination file must be a file, but " + dstFilePath +" is a folder");
+            } else if (!overwrite) {
+                logger.info("the destination file {} already exists, download operation will be ignored", dstFilePath);
+                return;
+            }
+        } else org.apache.commons.io.FileUtils.createParentDirectories(dstFile);
         S3Object o = s3Client.getObject(BUCKET_NAME, srcFilePath);
         try (S3ObjectInputStream s3is = o.getObjectContent();
-             FileOutputStream fos = new FileOutputStream(dstFile)) {
+             FileOutputStream fos = new FileOutputStream(dstFilePath)) {
```

Comment:
Please format this line.

Reply:
> Please format this line.

The code is already the same as the latest one you gave

Reply:
> Please format this line.

like this？
![image](https://user-images.githubusercontent.com/39169452/176811747-44931d40-9337-44cd-9eea-1564d721e7ff.png)

Reply:
Looks like you've solved it.